base/applications/mspaint/drawing.cpp

```diff
@@ -91,14 +91,35 @@ Poly(HDC hdc, POINT * lpPoints, int nCount,  COLORREF fg,  COLORREF bg, int thic
     SetROP2(hdc, oldRop);
 }
 
+double GetDistancePow2(POINT p1, POINT p2)
+{
+    LONG dx = p2.x - p1.x;
+    LONG dy = p2.y - p1.y;
+    return dx * dx + dy * dy;
+}
+
 void
 Bezier(HDC hdc, POINT p1, POINT p2, POINT p3, POINT p4, COLORREF color, int thickness)
 {
     HPEN oldPen;
     POINT fourPoints[4];
+    double distance12 = GetDistancePow2(p1, p2);
+    double distance13 = GetDistancePow2(p1, p3);
+    double distance42 = GetDistancePow2(p4, p2);
+    double distance43 = GetDistancePow2(p4, p3);
+    double norm1 = distance42 - distance12;
+    double norm2 = distance43 - distance13;
```

Comment:
Why using doubles? The GetDistancePow2 evaluated squared differences of LONG, so the result is naturally a LONG (if casted to a double, the decimal part would always be zero), and then you look at differences of distances, they will still be LONG.

Reply:
Fixed.